src/Interpreters/ClusterFunctionReadTask.cpp

```diff
@@ -61,6 +67,13 @@ void ClusterFunctionReadTaskResponse::serialize(WriteBuffer & out, size_t protoc
         else
             ActionsDAG().serialize(out, registry);
     }
+
+    if (protocol_version >= DBMS_CLUSTER_PROCESSING_PROTOCOL_VERSION_WITH_FILE_BUCKETS_INFO)
+    {
+        FormatFactory::instance().serializeFileFormatName(file_bucket_info, out);
```

Comment:
These methods `FormatFactory::serializeFileFormatName/deserializeFileFormatName` looks very strange and confusing to me. It will be more natural and readable to have a code like this:
```c++
if (file_bucket_info)
{
    /// Write format name so we can create appropriate file bucket info during deserialization.
    writeStringBinary(file_bucket_info->getFormatName(), out);
    file_bucket_info->serialize(out);
}
else
{
    /// Write empty string as format name if file_bucket_info is not set.
    writeStringBinary("", out);
}
```

And for deserialization:
```c++
String format;
readStringBinary(format, in);
if (!format.empty())
{
    file_bucket_info = FormatFactory::instance().getFileBucketInfo(format);
    file_bucket_info->deserialize(in);
}
```